ingester/src/init/wal_replay.rs

```
@@ -214,7 +214,12 @@ where
     let segment_id = batches.id();
 
     for batch in batches {
-        if let Err(err @ wal::Error::UnableToReadNextOps { .. }) = batch {
+        if let Err(
+            err @ wal::Error::UnableToReadNextOps {
+                source: wal::blocking::ReaderError::IncompleteEntry { .. },
```

With the new error disambiguation, WAL replay now only swallows the specific `IncompleteEntry` error. The tests below cover a set of `source` errors we wouldn't want to swallow when encountering `UnableToReadNextOps` to assert this behaviour.